Bump pyairnow to 1.3.1

natepugh opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Bump pyairnow lib version to 1.3.1.

[Changelog](https://github.com/asymworks/pyairnow/blob/master/CHANGELOG.md#130---2025-06-24)
[change diff](https://github.com/asymworks/pyairnow/compare/v1.2.1...v1.3.1)

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->
- This PR fixes or closes issue: fixes #140622

The pyairnow library was updated to reflect changes the EPA AirNow Technical Assistant document, including updating the aqi <-> pollutant concentration conversion methods to handle AQI values beyond 500. See the [pyairnow/pull/11](https://github.com/asymworks/pyairnow/pull/11) for more detail.

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed: N/A

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

natepugh committed:
Bump pyairnow to 1.30.0 #140622

natepugh committed:
Merge branch 'dev' into airnow-issue-140622

abmantis reviewed (changes requested):
Please add the diff/compare link or changelog (for all intermediate versions), as mentioned in point 7 of https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

abmantis commented:
> > Please add the diff/compare link or changelog (for all intermediate versions), as mentioned in point 7 of https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr
> 
> Thanks for your review! I've updated the body of the PR with the [Changelog](https://github.com/asymworks/pyairnow/blob/master/CHANGELOG.md) and [diff link](https://github.com/asymworks/pyairnow/compare/v1.2.2...v1.3.0)

Thanks! I've updated the diff link, since it was comparing from 1.2.2 instead of 1.2.1

abmantis commented:
The CI is failing :/ Can you check it out?

natepugh commented:
I'm curious about what the right process is here- the failing mypy checks are unrelated to the library version bump, but I'm sure _should_ be taken care of. Do we need to make that another PR?

natepugh committed:
Fix mypy test failures.

natepugh commented:
`pyairnow` has been updated to release 1.3.1. I've updated this PR to reflect that, and fixed the mypy issue in this repo that was preventing a merge.

abmantis committed:
Update homeassistant/components/airnow/coordinator.py

abmantis commented:
Thanks @natepugh !

natepugh commented:
Thanks!
